### PR TITLE
[ON-HOLD] Add function to get container memory limit

### PR DESCRIPTION
**Pull request by tmathern**
## Description
Add function to get container memory limit, so the underlying file where the limit is can easily be read (especially since this library already reads it).

## Related Issue 
DEVOPS

## Motivation and Context

Add function to get container memory limit, so the underlying file where the limit is can easily be read (especially since this library already reads it). Needed for debugging OOM exceptions,

## How Has This Been Tested?

Unit tests, existing and new ones.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/memory.js [hunk 1]
```diff
@@ -37,18 +37,32 @@ function containerUsagePercentage(memoryContainerUsage=false){
...
 	}
-	return (memoryContainerUsage / limit) * 100;
+	return limit;
 }
 
 module.exports = {
 	containerUsage,
-	containerUsagePercentage
+	containerUsagePercentage,
+    containerMemoryLimit
```
Comment (adamcin) [inline lib/memory.js:67]: tabs in this file

Comment (tmathern) [reply lib/memory.js:67]: good catch. Fixed all formatting in all files.

File: lib/memory.js [hunk 2]
```diff
@@ -18,37 +18,51 @@ const UNLIMITED_MEMORY_AMOUNT = 9223372036854771712;
...
+    const limit = containerMemoryLimit(memoryContainerUsage);
+    return (memoryContainerUsage / limit) * 100;
+}
+
+/**
+ * Reads metrics from `/sys/fs/cgroup/memory`
+ * @returns {Number} container memory limit (bytes)
+ */
+function containerMemoryLimit(memoryContainerUsage = false) {
+    if (!(memoryContainerUsage)) {
```
Comment (adamcin) [inline lib/memory.js:49]: I don't think this block or the memoryContainerUsage parameter are needed here

Comment (tmathern) [reply lib/memory.js]: Correct again!

**Review by adamcin — APPROVED** [review]
LGTM

File: README.md [hunk 3]
```diff
@@ -9,8 +9,8 @@ Node Module for reading [cgroup](https://www.kernel.org/doc/Documentation/cgroup
 
 Raw values:
 - `stat.rss`: # of bytes of anonymous and swap cache memory
-- `kmem.usage_in_bytes`: current kernel memory allocation
-- `limit_in_bytes`: limit of memory usage
+- `kmem.usage_in_bytes`: current kernel memory allocation in bytes
+- `limit_in_bytes`: limit of memory usage in bytes
```
Comment (jdelbick) [inline README.md:13]: the title of variable says its in bytes so I didn't think it was necessary lol

File: lib/cpu.js [hunk 4]
```diff
@@ -52,14 +52,14 @@ function stat() {
...
 }
 
 
 module.exports = {
-	usage,
-	stat,
-	calculateUsage
+    usage,
+    stat,
+    calculateUsage
```
Comment (jdelbick) [inline lib/cpu.js:64]: blame my old IDE settings from 2018 for this spacing lol

**Review by jdelbick — APPROVED** [review]
LGTM,
next time please try to do the refactoring/style changes in a different PR so its easier to review

File: test/metrics.test.js [hunk 5]
```diff
@@ -153,7 +153,7 @@ describe('cgroup Metrics', function() {
         console.log(`CPU user count: ${metrics_object.cpuacct.stat.user}`);
         console.log(`CPU system count: ${metrics_object.cpuacct.stat.system}`);
         assert.equal(metrics_object.memory.containerUsage, 6666);
-        assert.equal(metrics_object.memory.containerUsagePercentage, 6666/9999 * 100);
+        assert.equal(metrics_object.memory.containerUsagePercentage, 6666 / 9999 * 100);
```
Comment (jdelbick) [inline test/metrics.test.js:156]: nice to have, add check for mem limit as part of all the memory metrics:
```
        assert.equal(metrics_object.memory.containerMemoryLimit, 9999);
```

**Comment by jdelbick** [comment]
also, please add the new metric function to the readme usage: https://github.com/adobe/node-cgroup-metrics#memory-metrics-1

File: test/metrics.test.js (hunk 5 above)
Comment (tmathern) [reply test/metrics.test.js:156]: are they all automatically included? Because I put tests in the memory.test.js file instead.

**Comment by tmathern** [comment]
on hold for now...

**Pull request closed without merging**